core/src/main/scala/cell/Cell.scala

```diff
@@ -464,7 +464,12 @@ private class CellImpl[K <: Key[V], V](pool: HandlerPool, val key: K, lattice: L
     // the only call to `tryCompleteAndGetState`
     val res = tryCompleteAndGetState(resolved) match {
       case finalRes: Try[_] => // was already complete
-        val res = finalRes == value // FIXME: should compare to joined value
+        val finalResult = finalRes.asInstanceOf[Try[V]].get
+        val newVal = value.map(tryJoin(finalResult, _))
+        val res = finalRes == newVal
+        if (!res) {
+          println(s"problem with $this; existing value: $finalRes, new value: $newVal")
```

Comment:
This output should be replaced with an exception. There should also be a new test, verifying the presence and/or absence of the exception.